### PR TITLE
E2E tests for kubectl logs and patch

### PR DESCRIPTION
**Pull request by mwielgus**
ref #10764

### CODE REVIEW

**Comment by thockin** [comment]
Ideas for a reviewer?

File: test/e2e/kubectl.go [hunk 1]
```diff
@@ -238,6 +238,51 @@ var _ = Describe("Kubectl client", func() {
...
 
+	Describe("Kubectl logs", func() {
+		It("should find a string in pod logs", func() {
+			mkpath := func(file string) string {
+				return filepath.Join(testContext.RepoRoot, "examples/guestbook-go", file)
+			}
+			controllerJson := mkpath("redis-master-controller.json")
+			nsFlag := fmt.Sprintf("--namespace=%v", ns)
+			By("creating Redis RC")
+			runKubectl("create", "-f", controllerJson, nsFlag)
```
Comment (jlowdermilk) [inline test/e2e/kubectl.go]: Add a call to `cleanup` in defer, so we aren't leaking resources. Same for `Kubectl patch` below.

Comment (mwielgus) [inline test/e2e/kubectl.go]: The namespace is deleted after each test:
https://github.com/mwielgus/kubernetes/blob/kubectl_e2e_patch/test/e2e/kubectl.go#L72

Comment (jlowdermilk) [inline test/e2e/kubectl.go]: That works then.

**Comment by jlowdermilk** [comment]
LGTM

**Pull request merged**